### PR TITLE
download pipeline and/or file list files from online

### LINKED ISSUE #3340: Allow for URLs as argument with -p switch
If pipelines are served up through URLs (e.g., via S3), it would be nice for CP to accept them through the `-p` switch.

(If already implemented, then please close)

### PR DESCRIPTION
**Pull request by bethac07**
Resolves #3340 and #3341 

We may or may not want to actually pull this, since we seem to be solving it differently in master, but this is a fix that would work for 3.1.9 and backwards

### CODE REVIEW

File: cellprofiler/__main__.py [hunk 1]
```diff
@@ -69,7 +70,8 @@ def main(args=None):
 
     options, args = parse_args(args)
 
-    if any(options.pipeline_filename.startswith(protocol) for protocol in ('http', 'https', 'ftp')):
+    o = urlparse.urlparse(options.pipeline_filename)
```
Comment (braymp) [inline cellprofiler/__main__.py]: Need to add `if options.pipeline_filename:` around lines 73-81, or else NoneType error results.

File: cellprofiler/__main__.py [hunk 2]
```diff
@@ -69,7 +70,8 @@ def main(args=None):
...
+    if o[0] in ("ftp", "http", "https"):
         import urllib2
         temp_pipe_file = tempfile.NamedTemporaryFile(mode='w+b',suffix='.cppipe')
```
Comment (braymp) [inline cellprofiler/__main__.py]: Need to add something like the below before the `temp_pipe_file` assignment, or else an error can occur if the default temp location doesn't exist:
```
if options.temp_dir is not None:
    if not os.path.exists(options.temp_dir):
        os.makedirs(options.temp_dir)
        temp_dir = options.temp_dir
    else:
        temp_dir = cpprefs.get_temporary_directory()
```

File: cellprofiler/__main__.py [hunk 3]
```diff
@@ -69,7 +70,8 @@ def main(args=None):
 
     options, args = parse_args(args)
 
-    if any(options.pipeline_filename.startswith(protocol) for protocol in ('http', 'https', 'ftp')):
+    o = urlparse.urlparse(options.pipeline_filename)
+    if o[0] in ("ftp", "http", "https"):
         import urllib2
```
Comment (braymp) [inline cellprofiler/__main__.py]: Need to add something like the below before the `temp_pipe_file` assignment, or else an error can occur if the default temp location doesn't exist:
```
if options.temp_dir is not None:
    if not os.path.exists(options.temp_dir):
        os.makedirs(options.temp_dir)
        temp_dir = options.temp_dir
    else:
        temp_dir = cpprefs.get_temporary_directory()
```

File: cellprofiler/__main__.py [hunk 4]
```diff
@@ -78,23 +80,27 @@ def main(args=None):
                 a.write(line)
```
Comment (braymp) [inline cellprofiler/__main__.py]: Changed 'with' loop to the following to deal with "IOError: [Errno 13] Permission denied" error; see https://stackoverflow.com/a/23212515
```
for line in downloaded_pipeline:
    temp_pipe_file.write(line)
```

**Comment by bethac07** [comment]
Closing this here in favor of doing this in Nucleus

**Pull request closed without merging**